include/hipSYCL/sycl/detail/device_array.hpp

```diff
@@ -42,37 +42,6 @@ struct device_array
   using iterator = T*;
   using const_iterator = const T*;
 
-#ifdef HIPSYCL_PLATFORM_HCC
-  // There seem to be problems with aggregate initialization with hcc?
-  template<size_t n = N,
-           std::enable_if_t<n == 1>* = nullptr>
-  HIPSYCL_UNIVERSAL_TARGET
-  explicit device_array(T x)
-    : _data{x}
-  {}
-
-  template<size_t n = N,
-           std::enable_if_t<n == 2>* = nullptr>
-  HIPSYCL_UNIVERSAL_TARGET
-  device_array(T x, T y)
-    : _data{x,y}
-  {}
-
-  template<size_t n = N,
-           std::enable_if_t<n == 3>* = nullptr>
-  HIPSYCL_UNIVERSAL_TARGET
-  device_array(T x, T y, T z)
-    : _data{x,y,z}
-  {}
-
-  HIPSYCL_UNIVERSAL_TARGET
-  device_array()
-    : _data{}
-  {}
-
-#endif
-
-
```

Comment:
Removing these constructors gives the same exact construction / assignment behavior as `std::array` while using a C-style array as the backing storage. Is this workaround still relevant? Did hcc maybe just expect double-brackets `{{x, y}}` for initialization of a `device_array`, like it was required for `std::array` before C++14?

If this issue cannot be solved without declaring a constructor, we should replace it by a variadic constructor like @arkantos493 suggested.

Reply:
We don't rely on hcc anymore and now use clang exlusively, so likely we can get rid of the workaround.